solutions/search/esql-full-text-filter-tutorial.md

```diff
@@ -0,0 +1,412 @@
+POST /_query?format=txt
+{
+  "query": """
+    FROM cooking_blog
+    | WHERE description:"fluffy pancakes"
+    | LIMIT 1000
+  """
+}
+```
+
+By default, like the Query DSL `match` query, ES|QL uses `OR` logic between terms. This means it will match documents that contain either "fluffy" or "pancakes", or both, in the description field.
+
+:::{tip}
+You can control which fields to include in the response using the `KEEP` command:
+
+```esql
+POST /_query?format=txt
+{
+  "query": """
+    FROM cooking_blog
+    | WHERE description:"fluffy pancakes"
+    | KEEP title, description, rating
+    | LIMIT 1000
+  """
+}
+```
+:::
+
+### Require all terms in a match query
+
+Sometimes you need to require that all search terms appear in the matching documents. Here's how to do that using the function syntax with the `operator` parameter:
+
+```esql
+POST /_query?format=txt
+{
+  "query": """
+    FROM cooking_blog
+    | WHERE match(description, "fluffy pancakes", {"operator": "AND"})
+    | LIMIT 1000
+  """
+}
+```
+
+This stricter search returns *zero hits* on our sample data, as no document contains both "fluffy" and "pancakes" in the description.
+
+### Specify a minimum number of terms to match
+
+Sometimes requiring all terms is too strict, but the default OR behavior is too lenient. You can specify a minimum number of terms that must match:
+
+```esql
+POST /_query?format=txt
+{
+  "query": """
+    FROM cooking_blog
+    | WHERE match(title, "fluffy pancakes breakfast", {"minimum_should_match": 2})
+    | LIMIT 1000
+  """
+}
+```
+
+This query searches the title field to match at least 2 of the 3 terms: "fluffy", "pancakes", or "breakfast".
+
+## Step 4: Search across multiple fields at once
+
+When users enter a search query, they often don't know (or care) whether their search terms appear in a specific field. ES|QL provides ways to search across multiple fields simultaneously:
+
+```esql
+POST /_query?format=txt
+{
+  "query": """
+    FROM cooking_blog
+    | WHERE title:"vegetarian curry" OR description:"vegetarian curry" OR tags:"vegetarian curry"
+    | LIMIT 1000
+  """
+}
+```
+
+This query searches for "vegetarian curry" across the title, description, and tags fields. Each field is treated with equal importance.
+
+However, in many cases, matches in certain fields (like the title) might be more relevant than others. We can adjust the importance of each field using scoring:
+
+```esql
+POST /_query?format=txt
+{
+  "query": """
+    FROM cooking_blog METADATA _score
+    | WHERE match(title, "vegetarian curry", {"boost": 2.0}) 
+        OR match(description, "vegetarian curry") 
+        OR match(tags, "vegetarian curry")
+    | KEEP title, description, tags, _score
+    | SORT _score DESC
+    | LIMIT 1000
+  """
+}
+```
+
+In this example, we're using the `boost` parameter to make matches in the title field twice as important as matches in other fields. We also request the `_score` metadata field to sort results by relevance.
```

Comment:
I think it's important to clarify how _score works:

- No score? You're filtering, not caring about relevance
- Scoring? Then, take into account:
  - Full text functions affect scoring
  - Other things don't (this is not true for 8.18 as there, other things provide a constant score, but will be true in 8.19 / 9.x)
  - You need to sort by _score as mentioned in the example

Reply:
Cool I'll add a tip about this, and we should go into detail on the metadata `_score` entry eventually in https://www.elastic.co/guide/en/elasticsearch/reference/current/esql-metadata-fields.html

@carlosdelest let me know if I can help out on that too :)

Reply:
Done!

Reply:
@carlosdelest this PR is targeting 9.0 docs. does this caveat about "other things provide a constant score" still true in 9.0?

Reply:
See https://github.com/elastic/docs-content/pull/809/commits/22b1ad171b3a8b126218e2e74507b64796e9f316

Reply:
Oh, you're right - 9.x will have that fixed. Sorry for the noise 😢

Reply:
cool no worries thanks for clearing that up, I can make this simpler so 👍